topics/functions-intro.md

```diff
@@ -24,68 +27,54 @@ This approach suits many light-weight scripting use cases, but introduces severa
-An execution engine that is part of the definition of the function handles running it.
-An engine can theoretically execute functions in any language as long as it respects several rules (such as the ability to terminate an executing function).
-
-Presently, as noted above, Valkey ships with a single embedded [Lua 5.1](lua-api.md) engine.
-There are plans to support additional engines in the future.
-Valkey functions can use all of Lua's available capabilities to ephemeral scripts,
-with the only exception being the [Valkey Lua scripts debugger](ldb.md).
+They can be loaded at startup and be used repeatedly by various applications and clients.
+Functions are also persisted to the AOF file and replicated from primary to replicas, so they are as durable as the data itself.
+When Valkey is used as an ephemeral cache, additional mechanisms (described below) are required to make functions more durable.
 
 Functions also simplify development by enabling code sharing.
-Every function belongs to a single library, and any given library can consist of multiple functions.
+Every function has a user-defined name and belongs to a library, and a library can consist of multiple functions.
 The library's contents are immutable, and selective updates of its functions aren't allowed.
 Instead, libraries are updated as a whole with all of their functions together in one operation.
 This allows calling functions from other functions within the same library, or sharing code between functions by using a common code in library-internal methods, that can also take language native arguments.
 
-Functions are intended to better support the use case of maintaining a consistent view for data entities through a logical schema, as mentioned above.
-As such, functions are stored alongside the data itself.
-Functions are also persisted to the AOF file and replicated from primary to replicas, so they are as durable as the data itself.
-When Valkey is used as an ephemeral cache, additional mechanisms (described below) are required to make functions more durable.
-
 Like all other operations in Valkey, the execution of a function is atomic.
 A function's execution blocks all server activities during its entire time, similarly to the semantics of [transactions](transactions.md).
 These semantics mean that all of the script's effects either have yet to happen or had already happened.
 The blocking semantics of an executed function apply to all connected clients at all times.
 Because running a function blocks the Valkey server, functions are meant to finish executing quickly, so you should avoid using long-running functions.
 
+Functions are written in [Lua 5.1](lua-api.md).
+Valkey functions can use all of Lua's available capabilities to ephemeral scripts,
+with the only exception being the [Valkey Lua scripts debugger](ldb.md).
+In the future, it may be possible to extend Valkey to allow other languages.
```

Comment:
I kind of don't like statements like this the docs: "in the future, it may be possible". 

They're had to maintain because if we do (or do not) support this, it'd be easy to overlook and give people wrong information (or false hope).

Reply:
Yeah, I wanted to delete it, but there is some implicit reference below about the potential existence of other "engines" which doesn't make much sense if we don't mention anything.

Do you have a better suggestion?

Reply:
Are you referring to "When the target execution engine handles the FUNCTION LOAD command, it registers the library's functions."?

I'm not too alarmed by this on it's own, but it could be revised to by changing "target" to "Lua".

Reply:
```suggestion
```

Reply:
I think we can ignore "target execution engine" for now. The module API for additional engines is close to being merged.